### PR TITLE
make configureFromProviderConfig more readable

### PR DESCRIPTION
**Pull request by ramaraochavali**
Make this function more reabable by moving `buildHCMTracing` out of switch case and only build what is necessary in switch case

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/tracing.go [hunk 1]
```diff
@@ -93,8 +94,8 @@ func configureTracingFromSpec(
 		}
 		return nil, nil
 	}
-
 	spec := tracing.ServerSpec
+	proxyCfg := proxy.Metadata.ProxyConfigOrDefault(push.Mesh.DefaultConfig)
```
Comment (zirain) [inline pilot/pkg/networking/core/v1alpha3/tracing.go:98]: Duplicate with L83?

**Review by zirain — COMMENTED** [review]
lgtm

File: pilot/pkg/networking/core/v1alpha3/tracing.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/tracing.go]: yeah. removed

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**